streams/jvm/src/main/scala/zio/stream/experimental/StringDecoder.scala

```diff
@@ -0,0 +1,229 @@
+package zio.stream.experimental
+
+import java.nio.charset.Charset
+import java.nio.charset.StandardCharsets
+import zio.Chunk
+
+object StringDecoder {
+
+  implicit class ByteStreamDecode[R, E, A](val stream: ZStream[R, E, Byte]) extends AnyVal {
```

Comment:
We should probably just add these methods on ZStream itself if we want infix versions of these. Implicit classes are unfortunately not very discoverable.

Pinging @adamgfraser for thoughts around API.

Reply:
Directly on ZStream these methods will have very fancy type signatures, while this might be a bit less discoverable but as simple as it gets.

Reply:
You mean in terms of the implicit evidence? That's fine- many of the ZStream methods already require implicit evidence. It's preferred in the project because of the much better discoverability.